### PR TITLE
Upgrade Go to 1.20.2

### PR DESCRIPTION
**Pull request by braydonk**
## Description
This PR upgrades the version of Go required by the Ops Agent and updates build containers to install go1.20.2.

## Related issue
b/274799145

## How has this been tested?
Ran `make test` and `make build` after changing everything.

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -42,9 +42,9 @@ RUN set -x; yum -y update && \
 SHELL ["/bin/bash", "-c"]
 
 # Install golang
-ADD https://golang.org/dl/go1.19.linux-amd64.tar.gz /tmp/go1.19.linux-amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (braydonk) [reply Dockerfile:45]: Does `dockerfiles/template` count? This used to be a manual update throughout the whole Dockerfile, however the new tool to compile a Dockerfile from the teamplate did mean I only had to manually fix it there and then recompile the Dockerfile. Did you have something else in mind?

File: integration_test/ops_agent_test.go [hunk 2]
```diff
@@ -28,13 +28,16 @@ The following variables are optional:
 
 REPO_SUFFIX: If provided, what package repo suffix to install the ops agent from.
 ARTIFACT_REGISTRY_REGION: If provided, signals to the install scripts that the
-    above REPO_SUFFIX is an artifact registry repo and specifies what region it
-	is in. If not provided, that means that the packages are accessed through
-	packages.cloud.google.com instead, which may point to Cloud Rapture or
-	Artifact Registry under the hood.
+
```
Comment (braydonk) [inline integration_test/ops_agent_test.go:31]: `gofmt` did this automatically. I think it's a `godoc` convention. Maybe we should reformat this comment if we don't want it to look like this?

**Review by rafaelwestphal — APPROVED** [review]

**Review by avilevy18 — APPROVED** [review]

**Comment by braydonk** [comment]
Due to some weirdness building the HTTP server for Prometheus tests with Go 1.20, going to push that upgrade to a separate bug and PR. Building the Integration Test applications with Go 1.19 shouldn't have any negative effects with the Ops Agent being built in Go 1.20; they are indpendent.

**Comment by braydonk** [comment]
Going to re-run, there is a small chance that SAP Hana failure is real but I think it's most likely a flake.

**Comment by braydonk** [comment]
Okay, SAP Hana was indeed a flake, and I'm pretty confident those Windows failures are also flakes because they passed before on the same commit. Going to merge.

**Pull request merged**